java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -370,20 +367,12 @@ public void allocateNew() {
    */
   @Override
   public boolean allocateNewSafe() {
-    checkDataBufferSize(lastValueAllocationSizeInBytes);
-    computeAndCheckOffsetsBufferSize(lastValueCapacity);
-
-    /* we are doing a new allocation -- release the current buffers */
-    clear();
-
     try {
-      allocateBytes(lastValueAllocationSizeInBytes, lastValueCapacity);
+      allocateNew(lastValueAllocationSizeInBytes, lastValueCapacity);
+      return true;
     } catch (Exception e) {
-      clear();
```

Comment:
why isn't clear necessary anymore?

Reply:
because allocateNew is already doing the same. please see line 410..